### DISCUSSION TITLE
feat: data edit and confirmation screen

### DISCUSSION THREAD

**Discussion by sercanerhan** [acikkaynak/deprem-yardim-frontend#706] [Ideas]
>The following items contain assumptions and need to be validated by (especially by the data leads).
## WHY

Data cannot be imported into the DB and displayed with a delay in the application/live until the data editing and validation processes, which the data team has done with manual controls over the [sample google sheet](https://docs.google.com/spreadsheets/d/1SKM5pMoA6mvI00ia0nzNnxIkLgNsJMXlhO2jgrwWucc/edit#gid=975519625), is completed. By minimizing this delay; Apart from bulk data transfer, we are looking for a solution that can instantly reflect the controlled data to the application.

## HOW
To check the relevant row in the data control:

- Address search via Google Maps & Yandex Maps (or XXX website if needed)
- Twitter Search
It makes various searches on the relevant data/event, especially on the subject.
If it can display these basic functions specific to a single notification on the same web page, it will be able to control data much faster, and it will be able to flag this UNCONFIRMED data served over the DB as APPROVED via the DB after its operation.

This will allow the live application to instantly display the approved data.

## WHAT
What to do can be summarized under the following headings:
* Data editing / Confirmation UI/UX
* Login & Authorization
     * Edit & Approval powers
* Get Update Endpoints
* A record that is currently edited does not fall into someone else (So more than one person does not work with the same record. Returning a record that is not currently locked from the GET method)
* Search by ID
* Updated data to be visible in the application

## Risks
* While bulk editing on GSheet can be done easily when needed, it needs to be updated via DB now. The resource that will do the work changes.
* Is it necessary to version the data against data loss? So person X approved or made an edit, then **v1**. Is it necessary to go back to **v0** afterward?

** discord kullanıcı adı:**  @sercanerhan#0543

![data edit and confirmation screen](https://user-images.githubusercontent.com/3420568/218209489-7829b010-27b1-46d6-b59d-42a78908ee92.jpg)

## Değerlendirdiğiniz alternatifleri tanımlayın
If the assumption presented for the main problem above is accepted, with this screen to be provided, we will have the ability to display instantaneously verified data, which I believe is a valuable gain.

## Ek Bağlam
In order not to waste time, I am sharing the draft mock-up (!) that I drew very badly.

**Comment by yusufgns** [comment]
Issues başlığınızı ilgili Issues type ekliyerek düzenler misiniz?
https://github.com/acikkaynak/deprem-yardim-frontend/tree/development/.github/ISSUE_TEMPLATE

**Comment by sercanerhan** [comment]
teşekkür ederim uyarın için @yusufgns :pray:

**Comment by yusufgns** [comment]
İçerik yazınızda bazı hatalar mevcut, discord isminiz image altına sarkmış durumda. Labels ekliyorum düzeltirsiniz.

**Comment by sercanerhan** [comment]
yazı yazmayı öğrenirsem, issue'da açacağım :(  sağolasın tekrar :pray:

**Comment by eraygundogmus** [comment]
@usirin bu bizle alakalı mı?

**Comment by sercanerhan** [comment]
Uygun görülürse ister, design talebimiz olacaktır  @eraygundogmus  evet.

**Comment by usirin** [comment]
Discussion'a ceviriyorum bunu, sonuc cikinca frontend tarafinda yapilmasi gerekenlerle ilgili yeni bi issue acariz.

**Comment by sercanerhan** [comment]
https://deprem-adres-kontrolleri.vercel.app/ ile de kesişiyor sanırım?
